### PR TITLE
Implement GSFont.axes

### PR DESCRIPTION
**Pull request by madig**
This PR goes towards supporting the Axes font-level parameter (https://github.com/googlefonts/glyphsLib/issues/585). I want to use it later for determining what axes to write to the Designspace.

### CODE REVIEW

File: tests/classes_test.py [hunk 1]
```diff
@@ -183,6 +183,74 @@ def test_font_master_proxy(self):
...
+    def test_axes_parameter(self):
+        font = GSFont()
+        font.customParameters["Axes"] = [
+            {"Name": "Weight", "Tag": "wght"},
+            {"Name": "Empty Name", "Tag": "EMPT"},
+            {"Name": "Empty Name", "Tag": "EMPT"},
+            {"Name": "Empty Name", "Tag": "EMPT"},
+            {"Name": "Empty Name", "Tag": "EMPT"},
+            {"Name": "Empty Name", "Tag": "EMPT"},
+            {"Name": "Empty Name", "Tag": "EMPT"},
```
Comment (belluzj) [inline tests/classes_test.py]: What are these?

Comment (madig) [reply tests/classes_test.py]: The default strings if you repeatedly add axes in the "Axes" dialog but don't change them.

Comment (belluzj) [reply tests/classes_test.py]: Hm ok... can you write it down as a comment?

**Comment by madig** [comment]
Oops, I must also implement `GSFontMaster.axes`.

**Comment by khaledhosny** [comment]
Can we have [GSInstance.axes](https://docu.glyphsapp.com/#GSInstance.axes) as well?

**Comment by madig** [comment]
Yes, but this PR has been called into question by my co-maintainers. I was basically trying to solve axis handling with a clean approach but am now stalled by https://github.com/googlefonts/glyphsLib/issues/568#issuecomment-588256106.

**Comment by khaledhosny** [comment]
For my use case, I want whatever GlyphsApp would return for these properties, and looking into https://github.com/schriftgestalt/GlyphsSDK/blob/master/ObjectWrapper/GlyphsApp/__init__.py#L1633-L1753, I think what this PR is doing is pretty close. The complicated part is getting the font axes, but may be a solution that works reliably only when `Axes` custom parameter is set is better than nothing?

**Comment by madig** [comment]
If you define the Axes parameter, `GSFont.axes` returns it unchanged. You can use its' `len` to decide which master and instance values to consult.

**Comment by khaledhosny** [comment]
I do this already. It would be slightly nicer if I didn’t have to replicate this logic everytime I use glyphsLib (I’m using glyphsLib and fontTools to build fonts and bypass the UFO conversion done by fontmake).

**Comment by madig** [comment]
Oooh, do you have a glyphs2ft library?

**Comment by khaledhosny** [comment]
Not a library, some 400 lines of ugly project specific code, but it gives me total control instead of fighting both fontmake and Glyphs into doing the super weird things I want to build.

**Comment by madig** [comment]
I guess I can add a GSInstance.axes once my co-maintainers approve this one. Since I'm not sure which direction this lib is going to take in the future, I don't want to add Glyphs APIs I may not want later...

**Comment by schriftgestalt** [comment]
The Glyphs python API has GSFont.axes, GSFontMaster.axes and GSInstance.axes for some time now. Each returns a list for floats.

**Comment by khaledhosny** [comment]
Is there anything blocking the PR?

**Comment by madig** [comment]
Me not knowing if we want this. What was I even trying to solve?

**Comment by schriftgestalt** [comment]
It can be useful if glyphsLib is used for other things then just a gateway to ufo.

**Comment by khaledhosny** [comment]
Like I said, I do have a need for it so I don’t see the issue in merging it. Though the focus for GlyphLib is conversion to UFO, I often use it to read and manipulate Glyphs files without going through UFO and I don’t see why supporting convenient APIs would be something undesirable.

**Comment by khaledhosny** [comment]
Both GSFont and  GSInstance got axes attribute in the Glyphs 3 support work.

**Pull request closed without merging**